src/coreComponents/mesh/generators/CellBlockManagerABC.hpp

```diff
@@ -0,0 +1,187 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 Total, S.A
+ * Copyright (c) 2020-     GEOSX Contributors
+ * All right reservepd
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+#ifndef GEOSX_CELLBLOCKMANAGERABC_HPP
+#define GEOSX_CELLBLOCKMANAGERABC_HPP
+
+#include "dataRepository/Group.hpp"
+
+#include <map>
+
+namespace geosx
+{
+
+/**
+ * @brief Abstract base class for CellBlockManager.
+ */
+class CellBlockManagerABC : public dataRepository::Group
+{
+public:
+
+  /**
+   * @brief Constructor
+   * @param name The name of this Group.
+   * @param parent The parent Group.
+   */
+  CellBlockManagerABC( string const & name, Group * const parent ):
+    Group( name, parent )
+  {
+    // Left blank
+  }
+
+  virtual ~CellBlockManagerABC() = default;
+
+  /**
+   * @brief Maximum number of edges allowed (in memory) per each node.
+   * @return The number as an integer.
+   */
+  static constexpr int maxEdgesPerNode()
+  { return 200; }
```

Comment:
What is the motivation between putting `maxEdgesPerNode()` and `maxFacesPerNodes()` in two separate files (`CellBlockManagerABC.hpp` and `CellBlockManager.hpp`?

Reply:
`maxEdgesPerNode` is called from `EdgeManager` and `CellBlock*` stuff. That is, both sides of the abstraction.
While `maxFacesPerNode` is only called from `CellBlockManager`.
I wanted to reduce the API to the minimum, but it's true that some symmetry is sacrificed here. What's your opinion? Would you prefer symmetry back?

However, a followup is surely to build an allocation policy, so wave equation do not have to pay the extra cost of our tricky reservoir grids. When we do this (when?... never? 😉 ), this will move!